Automattic/wp-calypso
# Add domain pending renewal state

#### Changes proposed in this Pull Request

This PR adds the state to the domain assembler and uses it to show a pending renewal stat in the domain list and the domain management pages.

<img width="1054" alt="Screen Shot 2021-10-06 at 2 16 12 PM" src="https://user-images.githubusercontent.com/1379730/136260331-9f91ee0f-bd45-4ea7-add6-cc0d83244af7.png">

<img width="780" alt="Screen Shot 2021-10-06 at 2 24 08 PM" src="https://user-images.githubusercontent.com/1379730/136261310-f7503cc5-4225-42e2-8a4b-20a0c343bf59.png">


#### Testing instructions

Set the pending renewal state on a domain as specified in D67981-code.
Check the domain list and the domain management pages to make sure the pending renewal status is shown properly.
On the domain management page, make sure that the "Renew now for..." button is not shown.

Remove the renewal pending state and make sure the state shows as "Active" and the "Renew now for..." button is shown again.
